src/ui/public/system_api/system_api.js

```diff
@@ -0,0 +1,31 @@
+const SYSTEM_API_HEADER_NAME = 'kbn-system-api';
+
+export default function SystemApisProvider() {
+
+  /**
+   * Adds a custom header designating request as system API
+   * @param headers Object representing set of headers
+   * @return Object representing set of headers, with system API header added in
+   */
+  function addSystemApiHeader(headers) {
+    const newHeaders = headers;
+    newHeaders[SYSTEM_API_HEADER_NAME] = true;
+    return newHeaders;
+  }
+
+  /**
+   * Returns true if request is a system API request; false otherwise
+   *
+   * @param request Object Request object created by $http service
+   * @return true if request is a system API request; false otherwise
+   */
+  function isSystemApiRequest(request) {
+    return !!request.headers[SYSTEM_API_HEADER_NAME];
+  }
+
+  // Public interface
+  return {
+    addSystemApiHeader,
+    isSystemApiRequest
```

Comment:
It's not just you, but I do think one must consider context when making that decision rather than a blanket policy.  For example, `lodashMerge` instead of `merge` doesn't provide any useful information, whereas `isRequest` is downright misleading without `SystemApi` in the title.  The function isn't checking whether it's a request, it's checking specifically whether the given request is a system api request.